Can't resolve 'react/jsx-runtime' at task "teambit.preview/preview:GeneratePreview" of env "teambit.react/react"

A similar [issue](https://github.com/teambit/bit/issues/3745) was closed previously stating that v15 should fix most React-TS issues, but I currently have this error with v15.

### Description
Using a Button component from Material-UI V5 (5.0.0-alpha.33) with React 17 in a new component causes the error

The following errors were found while running the build pipeline
Failed task 1: "teambit.preview/preview:GeneratePreview" of env "teambit.react/react"
component: my-scope/ui/form
ModuleNotFoundError: Module not found: Error: Can't resolve 'react/jsx-runtime' in 'C:\Users\{my user}\AppData\Local\Bit\capsules\5d1c9b36215bfc76acf3092779d44e75808be12b\node_modules\@material-ui\core\Button'

### Specifications
* Bit version: 0.0.386
* Node version: 14.16.0
* npm / yarn version: npm 6.14.11 / yarn 1.22.10
* Platform: Windows 10 Pro

### Context and additional information
**Prerequisites:**

* Installed bit with bvm, here is the output of `bvm version`
![image](https://user-images.githubusercontent.com/23360637/117692916-7e273000-b18b-11eb-9ea2-f19e105516cd.png)

**Steps to reproduce:**

* Create an empty folder
* Inside the folder, use powershell terminal via Visual Studio Code v1.56.0 to run `bit init --harmony`
* In the generated workspace.jsonc uncomment "teambit.react/react": {} and set "teambit.dependencies/yarn" has the packageManager, make sure to save file.
* Run the following commands:
  * `bit install react --type peer`
  * `bit install react-dom --type peer`
  * `bit install @testing-library/react`
   The dependencies in workspace.jsonc are the following:
   ![image](https://user-images.githubusercontent.com/23360637/117693610-3654d880-b18c-11eb-8c1a-ae53216a3cea.png)
* Create a basic component with the following command : `bit create react-component ui/form`

At this point, using `bit start` we can see that the component is working as expected with no errors.

Let's add Material-UI V5 (5.0.0-alpha.33)

* Run the following commands:
  * `bit install @material-ui/core@next --type peer`
  * `bit install @emotion/react --type peer`
  * `bit install @emotion/styled --type peer`
   The dependecies in workspace.json are the following:
   ![image](https://user-images.githubusercontent.com/23360637/117694552-4620ec80-b18d-11eb-8de6-530aed8f39a2.png)
* In button.tsx file, add `import Button from '@material-ui/core/Button';` under react import and use `<Button>{text}</Button>` instead of the {text} by itself
* Run `bit compile`

At this point, using `bit show ui/form` outputs the following and `bit status` is ok.
![image](https://user-images.githubusercontent.com/23360637/117696032-df9cce00-b18e-11eb-81e8-6b90bae7cd3e.png)

Finally, running either `bit start` or `bit build` outputs an error _Module not found: Error: Can't resolve 'react/jsx-runtime'_
The `bit build ` is a bit more readable here's the output
![image](https://user-images.githubusercontent.com/23360637/117696827-bf214380-b18f-11eb-8714-c8a45e68a8f6.png)

I also tried forcing dependencies on the component by specifing them in workspace.jsonc like this
![image](https://user-images.githubusercontent.com/23360637/117697067-0d364700-b190-11eb-93d3-5f4e43594be6.png)

but the error was exactly the same.

Thanks for any help !

I have the same problem but with react-native environment.

I'm importing a react-native component into another react-native component. 

If i run `bbit tag` or `bbit build` it gives me the following output errors.

```bash
Failed task 1: "teambit.preview/preview:GeneratePreview" of env "teambit.react/react-native"
component: my-scope/ui/item-list@1.0.0
ModuleNotFoundError: Module not found: Error: Can't resolve 'react-native' in '/home/damien/Library/Caches/Bit/capsules/4365eea46008f190976a5d219194843ecb2690a6/my-scope_ui_item@1.0.0/dist'.
```

Otherwise if I add a single react-native component that not depends on another, it works.

I don't know if it needs another issue so I'm posting it here.
Bit version: 0.0.405

--- a/scopes/react/react/webpack/webpack.config.preview.dev.ts
+++ b/scopes/react/react/webpack/webpack.config.preview.dev.ts
@@ -258,6 +258,8 @@ export default function ({ envId, fileMapPath, workDir }: Options): WebpackConfi
 
       // this is for resolving react from env and not from consuming project
       alias: {
+        'react/jsx-dev-runtime': require.resolve('react/jsx-dev-runtime.js'),
+        'react/jsx-runtime': require.resolve('react/jsx-runtime.js'),
         react: require.resolve('react'),
         '@teambit/mdx.ui.mdx-scope-context': require.resolve('@teambit/mdx.ui.mdx-scope-context'),
         'react-dom/server': require.resolve('react-dom/server'),

--- a/scopes/react/react/webpack/webpack.config.preview.ts
+++ b/scopes/react/react/webpack/webpack.config.preview.ts
@@ -167,6 +167,8 @@ export default function (fileMapPath: string): Configuration {
       extensions: moduleFileExtensions.map((ext) => `.${ext}`),
 
       alias: {
+        'react/jsx-dev-runtime': require.resolve('react/jsx-dev-runtime.js'),
+        'react/jsx-runtime': require.resolve('react/jsx-runtime.js'),
         // Support React Native Web
         // https://www.smashingmagazine.com/2016/08/a-glimpse-into-the-future-with-react-native-for-web/
         // TODO: @uri please remember to remove after publishing evangelist and base-ui
@@ -176,6 +178,7 @@ export default function (fileMapPath: string): Configuration {
         'react-dom': require.resolve('react-dom'),
         'react-native': 'react-native-web',
         '@mdx-js/react': require.resolve('@mdx-js/react'),
+
         // Allows for better profiling with ReactDevTools
         ...(isEnvProductionProfile && {
           'react-dom$': 'react-dom/profiling',

